client/app/components/saveable-form/field.js

```diff
@@ -2,15 +2,15 @@ import Component from '@glimmer/component';
 
 export default class SaveableFormFieldComponent extends Component {
   get value() {
-    return this.args.saveableChanges[this.args.attribute];
+    return this.args.data[this.args.attribute];
   }
 
   set value(newValue) {
-    this.args.saveableChanges[this.args.attribute] = newValue;
+    this.args.data[this.args.attribute] = newValue;
   }
 
   get error() {
-    return this.args.submittableChanges[`error.${this.args.attribute}`];
+    return this.args.error[this.args.attribute];
```

Comment:
At this point maybe we can just treat error as a string? And not assume we need an error object? What other properties could the error object hold?

Reply:
@godfreyyeung 

Ah you mean pass the error prop? I'll try that but I'm not sure it'll work because the `attribute` key isn't always present in the error. But let me give it a shot because that would be a simpler design.

Reply:
@godfreyyeung oh, the attribute isn't know up front. So we need the full error object.

Reply:
👍

Reply:
@godfreyyeung  I don't follow. Do you mean `get @error @attribute`? If you're referring to parentField, it needs to have the same interface. We curry this component from saveableForm.

Reply:
approve?

Reply:
sorry was trying to wrap my head around the radio yielding